src/main/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/WebformServiceImpl.java

```diff
@@ -36,47 +34,54 @@
   private static final String TEMPLATE_CATEGORY = "respondent_category";
   private static final String TEMPLATE_DESCRIPTION = "respondent_description";
 
-  private EventPublisher eventPublisher;
-
   private NotificationClientApi notificationClient;
+  private CircuitBreaker webformCircuitBreaker;
+  private CircuitBreaker envoyCircuitBreaker;
 
   private AppConfig appConfig;
   @Autowired private RateLimiterClient rateLimiterClient;
-  @Autowired private CircuitBreaker circuitBreaker;
 
   /**
    * Constructor for WebformServiceImpl
    *
-   * @param eventPublisher service for publication of events to RabbitMQ
    * @param notificationClient Gov.uk Notify service client
+   * @param circuitBreaker circuit breaker
    * @param appConfig centralised configuration properties
    */
   @Autowired
   public WebformServiceImpl(
-      final EventPublisher eventPublisher,
       final NotificationClientApi notificationClient,
+      final @Qualifier("webformCb") CircuitBreaker webformCircuitBreaker,
+      final @Qualifier("envoyLimiterCb") CircuitBreaker envoyCircuitBreaker,
       final AppConfig appConfig) {
-    this.eventPublisher = eventPublisher;
     this.notificationClient = notificationClient;
+    this.webformCircuitBreaker = webformCircuitBreaker;
+    this.envoyCircuitBreaker = envoyCircuitBreaker;
     this.appConfig = appConfig;
   }
 
   @Override
-  public String sendWebformEvent(Webform webform) {
-
+  public UUID sendWebformEmail(WebformDTO webform) {
     checkWebformRateLimit(webform.getClientIP());
-
-    String transactionId =
-        eventPublisher.sendEvent(
-            EventType.WEB_FORM_REQUEST, Source.RESPONDENT_HOME, Channel.RH, webform);
-    return transactionId;
+    return this.webformCircuitBreaker.run(
```

Comment:
The circuit breaker stuff doesn't make it very obvious what's going on. 
So I think the code would benefit from a statement or 2 saying what the outcome of this method call is.
I'm assuming that the send is either successful in which case the method returns with a UUID, or if there is some failure (or the circuit breaker is open due to previous failures) that it fails with an exception. 
I guess that right now it's not at obvious exactly why there is a circuit breaker, so it would be good to capture that knowledge in the code.

Reply:
added javadoc